### ISSUE TITLE
Ingester crashes due to inactivity while reading messages off Kafka

### ISSUE THREAD

**Issue by ManjunathShivakumar** [jaegertracing/jaeger#2069]
## Requirement - what kind of business use case are you trying to solve?

Using Ingester 1.15.1 to read messages from Kafka Confluent Platform 5.0.0 and write them to Cassandra test schema.

## Problem - what in Jaeger blocks you from solving the requirement?

Ingester starts up fine. Once I start up the producer which publishes the traces in protobuf format, I can see the ingester reading the messages in the logs, but after a short period the deadlock detector kicks in and kills process, complaining of inactivity. Nothing gets processed or written to casssandra during entire process.

## Here are my startup setttings for ingester

```
SPAN_STORAGE_TYPE=cassandra
DEPENDENCY_STORAGE_TYPE=cassandra
LOG=debug
CASSANDRA_CLUSTER=ie1-xds03b-qa.qa.betfair
KAFKA_BROKERS=ie1-xkos01-qa.qa.betfair:9092,ie1-xkos02-qa.qa.betfair:9092,ie1-xkos03-qa.qa.betfair:9092
TOPIC=exchangeBetTracesManju
CASS_KEY_SPACE=jaeger_v1_test
KAFKA_ENCODING=protobuf
KAFKA_PROTOCOL_VERSION=2.0.0
/var/jdb/jaeger-1.15.1-linux-amd64/jaeger-ingester  --ingester.deadlockInterval=25s --log-level=$LOG --kafka.consumer.protocol-version=$KAFKA_PROTOCOL_VERSION  --kafka.consumer.client-id=jaeger-client-v1 --ingester.parallelism=100 --cassandra.span-store-write-cache-ttl=10s --cassandra.servers=$CASSANDRA_CLUSTER --kafka.consumer.brokers=$KAFKA_BROKERS --kafka.consumer.topic=$TOPIC --cassandra.keyspace=$CASS_KEY_SPACE --kafka.consumer.encoding=$KAFKA_ENCODING > /var/jdb/jaeger-1.15.1-linux-amd64/log-ingester.log 2>&1 &

```

### Here are ingester logs

```
2020/02/12 20:17:34 maxprocs: Leaving GOMAXPROCS=16: CPU quota undefined
{"level":"info","ts":1581538654.7382011,"caller":"flags/service.go:115","msg":"Mounting metrics handler on admin server","route":"/metrics"}
{"level":"info","ts":1581538654.7384086,"caller":"flags/admin.go:108","msg":"Mounting health check on admin server","route":"/"}
{"level":"info","ts":1581538654.73868,"caller":"flags/admin.go:114","msg":"Starting admin HTTP server","http-port":14270}
{"level":"info","ts":1581538654.7387073,"caller":"flags/admin.go:100","msg":"Admin server started","http-port":14270,"health-status":"unavailable"}
{"level":"info","ts":1581538654.7538068,"caller":"cassandra/factory.go:93","msg":"Cassandra archive storage configuration is empty, skipping"}
{"level":"debug","ts":1581538654.7630103,"caller":"consumer/deadlock_detector.go:151","msg":"Starting global deadlock detector"}
{"level":"info","ts":1581538654.7630513,"caller":"healthcheck/handler.go:128","msg":"Health Check state change","status":"ready"}
{"level":"info","ts":1581538654.763082,"caller":"consumer/consumer.go:78","msg":"Starting main loop"}
{"level":"info","ts":1581538654.8751671,"caller":"consumer/consumer.go:169","msg":"Starting error handler","partition":0}
{"level":"info","ts":1581538654.8752677,"caller":"consumer/consumer.go:112","msg":"Starting message handler","partition":0}
{"level":"info","ts":1581538654.8777602,"caller":"consumer/consumer.go:169","msg":"Starting error handler","partition":1}
{"level":"info","ts":1581538654.8777978,"caller":"consumer/consumer.go:112","msg":"Starting message handler","partition":1}
{"level":"info","ts":1581538654.8780355,"caller":"consumer/consumer.go:169","msg":"Starting error handler","partition":2}
{"level":"info","ts":1581538654.8780706,"caller":"consumer/consumer.go:112","msg":"Starting message handler","partition":2}
{"level":"debug","ts":1581538655.4692624,"caller":"consumer/consumer.go:141","msg":"Got msg","msg":{"Headers":[],"Timestamp":"0001-01-01T00:00:00Z","BlockTimestamp":"0001-01-01T00:00:00Z","Key":"AAAAAAAAAAA=","Value":"CjsKCHRyYWNlSWQxEgdzcGFuSWQxGgpvcGVyYXRpb24xMgcI6/my2IMuOgIIA1INCgt0ZXN0U2VydmljZQ==","Topic":"exchangeBetTracesManju","Partition":0,"Offset":7715}}
{"level":"info","ts":1581538655.4696288,"caller":"consumer/processor_factory.go:65","msg":"Creating new processors","partition":0}
{"level":"debug","ts":1581538655.4698048,"caller":"processor/parallel_processor.go:50","msg":"Spawning goroutines to process messages","num_routines":100}
{"level":"debug","ts":1581538656.4698534,"caller":"consumer/consumer.go:141","msg":"Got msg","msg":{"Headers":[],"Timestamp":"0001-01-01T00:00:00Z","BlockTimestamp":"0001-01-01T00:00:00Z","Key":"AAAAAAAAAAA=","Value":"CjsKCHRyYWNlSWQxEgdzcGFuSWQxGgpvcGVyYXRpb24xMgcI04Gz2IMuOgIIA1INCgt0ZXN0U2VydmljZQ==","Topic":"exchangeBetTracesManju","Partition":0,"Offset":7716}}
{"level":"debug","ts":1581538657.4702768,"caller":"consumer/consumer.go:141","msg":"Got msg","msg":{"Headers":[],"Timestamp":"0001-01-01T00:00:00Z","BlockTimestamp":"0001-01-01T00:00:00Z","Key":"AAAAAAAAAAA=","Value":"CjsKCHRyYWNlSWQxEgdzcGFuSWQxGgpvcGVyYXRpb24xMgcIvImz2IMuOgIIA1INCgt0ZXN0U2VydmljZQ==","Topic":"exchangeBetTracesManju","Partition":0,"Offset":7717}}
{"level":"debug","ts":1581538658.4709978,"caller":"consumer/consumer.go:141","msg":"Got msg","msg":{"Headers":[],"Timestamp":"0001-01-01T00:00:00Z","BlockTimestamp":"0001-01-01T00:00:00Z","Key":"AAAAAAAAAAA=","Value":"CjsKCHRyYWNlSWQxEgdzcGFuSWQxGgpvcGVyYXRpb24xMgcIpJGz2IMuOgIIA1INCgt0ZXN0U2VydmljZQ==","Topic":"exchangeBetTracesManju","Partition":0,"Offset":7718}}
{"level":"debug","ts":1581538679.4818714,"caller":"consumer/consumer.go:141","msg":"Got msg","msg":{"Headers":[],"Timestamp":"0001-01-01T00:00:00Z","BlockTimestamp":"0001-01-01T00:00:00Z","Key":"AAAAAAAAAAA=","Value":"CjsKCHRyYWNlSWQxEgdzcGFuSWQxGgpvcGVyYXRpb24xMgcIuLW02IMuOgIIA1INCgt0ZXN0U2VydmljZQ==","Topic":"exchangeBetTracesManju","Partition":0,"Offset":7739}}
{"level":"info","ts":1581538679.8780375,"caller":"consumer/consumer.go:155","msg":"Closing partition due to inactivity","partition":1}
{"level":"debug","ts":1581538679.8781123,"caller":"consumer/deadlock_detector.go:195","msg":"Closing deadlock detector","partition":1}
{"level":"warn","ts":1581538679.878072,"caller":"consumer/deadlock_detector.go:122","msg":"Signalling partition close due to inactivity","partition":1}
{"level":"info","ts":1581538679.8781605,"caller":"consumer/deadlock_detector.go:115","msg":"Closing ticker routine","partition":1}
{"level":"info","ts":1581538679.878181,"caller":"consumer/consumer.go:162","msg":"Closing partition consumer","partition":1}
{"level":"warn","ts":1581538679.8782256,"caller":"consumer/deadlock_detector.go:122","msg":"Signalling partition close due to inactivity","partition":2}
{"level":"info","ts":1581538679.8782465,"caller":"consumer/consumer.go:155","msg":"Closing partition due to inactivity","partition":2}
{"level":"debug","ts":1581538679.8782563,"caller":"consumer/deadlock_detector.go:195","msg":"Closing deadlock detector","partition":2}
{"level":"info","ts":1581538679.8782778,"caller":"consumer/consumer.go:162","msg":"Closing partition consumer","partition":2}
{"level":"info","ts":1581538679.8782873,"caller":"consumer/deadlock_detector.go:115","msg":"Closing ticker routine","partition":2}
{"level":"info","ts":1581538679.961061,"caller":"consumer/consumer.go:180","msg":"Finished handling errors","partition":1}
{"level":"info","ts":1581538679.9611216,"caller":"consumer/consumer.go:165","msg":"Closed partition consumer","partition":1}
{"level":"info","ts":1581538679.9837413,"caller":"consumer/consumer.go:180","msg":"Finished handling errors","partition":0}
{"level":"info","ts":1581538679.9837866,"caller":"consumer/consumer.go:138","msg":"Message channel closed. ","partition":0}
{"level":"debug","ts":1581538679.9838274,"caller":"processor/parallel_processor.go:75","msg":"Initiated shutdown of processor goroutines"}
{"level":"info","ts":1581538679.995977,"caller":"consumer/consumer.go:180","msg":"Finished handling errors","partition":2}
{"level":"info","ts":1581538679.9960089,"caller":"consumer/consumer.go:165","msg":"Closed partition consumer","partition":2}
{"level":"warn","ts":1581538704.875549,"caller":"consumer/deadlock_detector.go:122","msg":"Signalling partition close due to inactivity","partition":0}
.
.
.
panic: No messages processed in the last check interval

goroutine 73 [running]:
github.com/jaegertracing/jaeger/vendor/go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc0003a8370, 0xc0004c2200, 0x2, 0x2)
        /home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/go.uber.org/zap/zapcore/entry.go:230 +0x546
github.com/jaegertracing/jaeger/vendor/go.uber.org/zap.(*Logger).Panic(0xc000198c60, 0x1255799, 0x30, 0xc0004c2200, 0x2, 0x2)
        /home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/go.uber.org/zap/logger.go:225 +0x7f
github.com/jaegertracing/jaeger/cmd/ingester/app/consumer.newDeadlockDetector.func1(0xc0ffffffff)
        /home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/ingester/app/consumer/deadlock_detector.go:71 +0x40d
github.com/jaegertracing/jaeger/cmd/ingester/app/consumer.(*deadlockDetector).start.func1(0xc000122180, 0xc00000c6a0)
        /home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/ingester/app/consumer/deadlock_detector.go:163 +0x148
created by github.com/jaegertracing/jaeger/cmd/ingester/app/consumer.(*deadlockDetector).start
        /home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/ingester/app/consumer/deadlock_detector.go:152 +0x185
```

**Comment by yurishkuro** [comment]
you can disable deadlock detector via CLI flag by giving it 0 as run interval

**Comment by ManjunathShivakumar** [comment]
Thanks for the quick response @yurishkuro , I tried this. The basic problem is that the messages are not getting processed properly and not getting written to Cassandra, so if I disable the deadlock detector the process continues, but it does not make any progress. If I check the Prometheus metrics, the failed attempts counter keeps increasing, so there is some underlying problem processing the message.

I have enabled debug logs and can see the messages being logged in the ingester log, but how can I debug why they are not being processed.
